### PR TITLE
Polished content outline

### PR DESCRIPTION
**Pull request by rrrutledge**
* PR for the [Polished content outline](https://github.com/InnerSourceCommons/InnerSourceLearningPath/issues/56) issue.
* View the [rendered markdown](https://github.com/InnerSourceCommons/InnerSourceLearningPath/blob/contributor/outline/contributor/outline.md).

### CODE REVIEW

File: contributor/outline.md [hunk 1]
```diff
@@ -1,64 +1,138 @@
...
-    - In an Inner Source environment this might be less feasible due to a smaller amount of resources, however might make a pure research / r&d / maintenance transfer easier
+## Becoming an InnerSource Contributor
 
+  - A contributing mindset starts with a shared use mindset.
+    - When you have technical needs that are outside of the domain of your project,
+    look around to see if there is a shared solution that you can use.
+    - Briefly enumerate the benefits of using shared solutions.
+  - Using shared solutions puts you in a position to become a _Contributor_.
+    - There will be times when you wish that the shared solution has some new feature.
+    - Don't _Wait it Out_ or _Work Around_, spend your time and energy making the change directly in the shared project.
```
Comment (lenucksi) [inline contributor/outline.md]: Nice re-use of the Antipatterns. :) (We should probably also start writing them down as we do with the patterns I guess.)

File: contributor/outline.md [hunk 2]
```diff
@@ -1,64 +1,138 @@
...
+      - Knock on the door only when invited.
+    - With InnerSource projects there are several ways that code gets in.
+      - Sometimes you just open a pull request.
+      - At times you may need to get a feature request accepted before opening the pull request that fulfills it.
+      - You may need to talk to the host team in person before interacting with the project digitally.
+      In the same way that real-life hosts don't always appreciate a stranger knocking on their door, some InnerSource hosts appreaciate some heads-up before seeing a pull request get opened to their repo.
+      - In all of these interactions, be prepared to "sell" your contribution to the host team.
+        - Articulate the value that the contribution will bring to their ecosystem.
+        - Remind them that they are getting free engineering help with their project.
+        - Offer to fulfill a [30-day warranty] on your submission.
```
Comment (lenucksi) [inline contributor/outline.md]: Could hint here that this alleviates the fear-of-maintenance problem that might arise in the host team. (Advertisement for managers / host teams watching this segment) (Effectively a glimpse into one of the motivations for the pattern)

File: contributor/outline.md [hunk 3]
```diff
@@ -1,64 +1,138 @@
...
+  - Development
+    - If you get stuck, try to unblock yourself by reading the project documentation or looking at other parts of the code in the project.
+    - After trying the above if you're still stuck then reach out to the host team for help.
+    It doesn't do either side any good for you to spin forever on a problem that could be resolved in a few-minute conversation with someone more knowledgable about the project.
+    It's OK to ask for help.
+    You aren't expected to be as knowledgable and proficient as you might be on the full-time projects that your team owns.
+    - Match the existing coding style, indentation, etc, even if it's not what you would normally do.
+    A guest contribution for a feature is not the time to introduce a new way of structuring or formatting project code.
+  - Submitting the pull request
+    - Test out your contribution before opening the pull request.
```
Comment (lenucksi) [inline contributor/outline.md]: Oh yes please do so! ;) 
Would we want to hint to the value of CI/automation somewhere? Goes with the "can a contributor attempt to mentor host teams" bullet point that used to be in the outline - might actually be an example to explain this bullet point very briefly.

File: contributor/outline.md [hunk 4]
```diff
@@ -1,64 +1,138 @@
...
+    - High influence into the development, progression, and ecosystem of projects that you depend on.
+    - Relieved of the long-term burden of maintenance of your contributions.
+    The host team maintains them as part of maintaining the overall project.
+    - Key knowledge spread across multiple people and teams.
+    Improves the [bus factor].
+    - Do the right thing for the company by putting your work into shared projects that others can use instead of your own one-offs.
+  - Company
+    - Improved cross-company collaboration.
+    - Better overall project quality.
+    "With many eyes, all bugs are shallow"
```
Comment (lenucksi) [inline contributor/outline.md]: aka "Linus' Law"

File: contributor/outline.md [hunk 5]
```diff
@@ -1,64 +1,138 @@
...
+    - Reduced duplication of projects.
+    - Better alignment of shared projects to the needs of their internal consumers.
```
Comment (lenucksi) [inline contributor/outline.md]: We mention the career development, learning and fun at work effects for the individual, should we mention the potential improved people retention effect for the company here too?

File: contributor/outline.md (hunk 3 above)
Comment (rrrutledge) [reply contributor/outline.md]: Added a line that mentions running these automated checks.

There are some bullets in **The Contributor Ethos** that give specific examples of things that a _Contributor_ can do to mentor host teams on the value of letting them contribute.

I've intentionally left out anything about the _Contributor_ teaching them how to run their project better outside of their individual contribution.  I feel like that's a more advanced behavior and that this _Contributor_ section of tehe learning path should just teach a beginner how to get in their patch.

File: contributor/outline.md (hunk 4 above)
Comment (rrrutledge) [reply contributor/outline.md]: I didn't know that.

File: contributor/outline.md (hunk 3 above)
Comment (lenucksi) [reply contributor/outline.md]: Valid argument. Maybe we should start a list of things that could go in a 102 version of the learning path.

File: contributor/outline.md (hunk 4 above)
Comment (lenucksi) [reply contributor/outline.md]: https://en.wikipedia.org/wiki/Linus%27s_Law I argue that we should add the name and a / the link as a footnote. Adds legitimacy to the quote and credits the source of the phrase.

File: contributor/outline.md (hunk 3 above)
Comment (rrrutledge) [reply contributor/outline.md]: Open an issue in this project and just start putting comments on it?  We can decide what to do with them later.

**Comment by rrrutledge** [comment]
> I added two more questions

Great!  Incorporated.

**Comment by lenucksi** [comment]
Thanks for your approval @guywmartin, I'll merge it now.

**Comment by rrrutledge** [comment]
Thanks for these reviews.  I am still open to feedback and iteration on this PR from anyone that has it.

File: contributor/outline.md (hunk 3 above)
Comment (lenucksi) [reply contributor/outline.md]: Opened #62 for this

Comment (rrrutledge) [reply contributor/outline.md]: Thank you!

File: contributor/outline.md [hunk 6]
```diff
@@ -1,64 +1,141 @@
...
+    - Puts your development time into a place where others are also working.
+    It's not just up to you to maintain this area.
+    - Get help and support from the host team in implementing your changes.
+    - Your reputation and impact expands to outside your team.
+    - Learn from engineers on the host team and their coding and working styles.
+    This extra learning comes as part of your day job - no stretch or overtime assignment needed.
+    - Varied, fun, and personally fulfilling.
+  - Team
+    - You can get improvements you need in other projects on a timeline that you control.
+    - High influence into the development, progression, and ecosystem of projects that you depend on.
```
Comment (gruetter) [inline contributor/outline.md]: Whether or not the influence is high depends on your rank in the host project. It will be rather low in the beginning and might grow, but so does the commitment you need to make. I would rephrase this to a statement saying that this model is an _opportunity to influence_.

File: contributor/outline.md [hunk 7]
```diff
@@ -1,64 +1,141 @@
...
+    - Relieved of the long-term burden of maintenance of your contributions.
```
Comment (gruetter) [inline contributor/outline.md]: How about we s/Relieved of/Lower the/?

File: contributor/outline.md [hunk 8]
```diff
@@ -1,64 +1,141 @@
...
+    The host team maintains them as part of maintaining the overall project.
+    - Key knowledge spread across multiple people and teams.
+    Improves the [bus factor].
```
Comment (gruetter) [inline contributor/outline.md]: I think it _lowers_ the bus factor.

File: contributor/outline.md [hunk 9]
```diff
@@ -1,64 +1,141 @@
...
+    - Do the right thing for the company by putting your work into shared projects that others can use instead of your own one-offs.
```
Comment (gruetter) [inline contributor/outline.md]: This is a really important point, which we could stress somewhere else. I've seen this in our InnerSource initiative at Bosch a lot: people putting the well being of the company before that of their business unit or project. Maybe this is why higher management is much easier to convince than middle management.

And I guess this bullet point should be moved to the _Company_ section below

File: contributor/outline.md [hunk 10]
```diff
@@ -1,64 +1,141 @@
...
+  - Company
```
Comment (gruetter) [inline contributor/outline.md]: I propose to add _-more learning and knowledge sharing and thus more valuable employees_

File: contributor/outline.md [hunk 11]
```diff
@@ -1,64 +1,141 @@
...
+    - Improved cross-company collaboration.
+    - Better overall project quality.
```
Comment (gruetter) [inline contributor/outline.md]: not sure what _project quality_ is. Are you referring to _product quality_?

File: contributor/outline.md [hunk 12]
```diff
@@ -1,64 +1,141 @@
...
+    [Linus' Law](https://en.wikipedia.org/wiki/Linus%27s_Law): "With many eyes, all bugs are shallow"
+    - Reduced duplication of projects.
```
Comment (gruetter) [inline contributor/outline.md]: How about _Reduced waste induced by duplication of efforts and reinventing the wheel_, instead?

File: contributor/outline.md [hunk 13]
```diff
@@ -1,64 +1,141 @@
...
-   - Fixes for typos and builds are great contributions.
-   - Style of code should be similar to what code is already present.
+Outline for this section inspired by the [Trusted Committer introduction video].
 
-   - How acceptable / feasible / desired are upstream connected forks that work in a stable/experimental alternative setup? 
-    - The OSS environment has such forks where the experimental fork accepts more than the stable fork but there is exchange/upstreaming of proven experimental forks. (e.g. Psi/Psi+ or possibly mainline/LTS Linux)
-    - In an Inner Source environment this might be less feasible due to a smaller amount of resources, however might make a pure research / r&d / maintenance transfer easier
+## Becoming an InnerSource Contributor
 
+  - A contributing mindset starts with a shared use mindset.
```
[the remark this answers is not in the document]
Comment (rrrutledge) [reply contributor/outline.md]: Sure.

File: contributor/outline.md [hunk 14]
```diff
@@ -1,64 +1,141 @@
...
+    - When you have technical needs that are outside of the domain of your project,
```
[the remark this answers is not in the document]
Comment (rrrutledge) [reply contributor/outline.md]: Replaced _outside_ with _orthoganal_.  Does that make more sense?

File: contributor/outline.md [hunk 15]
```diff
@@ -1,64 +1,141 @@
...
+    look around to see if there is a shared solution that you can use.
+    - Briefly enumerate the benefits of using shared solutions.
+  - Using shared solutions puts you in a position to become a _Contributor_.
+    - There will be times when you wish that the shared solution has some new feature.
+    - Don't _Wait it Out_ or _Work Around_, spend your time and energy making the change directly in the shared project.
```
[the remark this answers is not in the document]
Comment (rrrutledge) [reply contributor/outline.md]: Done.

File: contributor/outline.md [hunk 16]
```diff
@@ -1,64 +1,141 @@
...
+    - Briefly enumerate the benefits of contributing.
+    This should be taken from the **Benefits** segment.
+    - You may need to spend a bit of time explaining these benefits to your management hierarchy in order to have their support to spend your time contributing.
```
[the remark this answers is not in the document]
Comment (rrrutledge) [reply contributor/outline.md]: Nice!  Just do it.  Updated to not make it sound like you're required to get manager permission to contribute.

File: contributor/outline.md [hunk 17]
```diff
@@ -1,64 +1,141 @@
...
+    - An InnerSource _Contributor_ knows that it doesn't matter where the code goes as long as they get the needed functionality.
```
[the remark this answers is not in the document]
Comment (rrrutledge) [reply contributor/outline.md]: Adjusted language.  What do you think?

File: contributor/outline.md [hunk 18]
```diff
@@ -1,64 +1,141 @@
...
+  - Contributing applies to all levels of project managements, not just code.
+    - Opening issues.
```
[the remark this answers is not in the document]
Comment (rrrutledge) [reply contributor/outline.md]: Done.

File: contributor/outline.md [hunk 19]
```diff
@@ -1,64 +1,141 @@
...
+    - Improving documention.
+    - Fixing builds.
```
[the remark this answers is not in the document]
Comment (rrrutledge) [reply contributor/outline.md]: Great!

File: contributor/outline.md [hunk 20]
```diff
@@ -1,64 +1,141 @@
...
+    - Anticipates and follows house rules.
+    - Understands they are not the home owner and acts accordingly.
+  - Let's talk about how each of these ideas applies in InnerSource.
+  - Getting in
+    - In the real world their are several ways that guests are invited in.
+      - An "open" sign lets them know they can walk right in without knocking.
+      - Knock on the door unexpected.
+      - Knock on the door only when invited.
+    - With InnerSource projects there are several ways that code gets in.
+      - Sometimes you just open a pull request.
```
[the remark this answers is not in the document]
Comment (rrrutledge) [reply contributor/outline.md]: I think so.  That idea seems to fit within this existing outline.  Does the outline here need to change?

File: contributor/outline.md [hunk 21]
```diff
@@ -1,64 +1,141 @@
...
+    Be sure to understand and follow them.
+    - If you are contributing the fix to an issue, be sure to follow all guidance and instruction documented in the issue.
+    - You should understand the project well enough to know that your contribution makes sense overall and is not just your personal snowflake.
+  - Understands they are not the home owner and acts accordingly.
+    - Leave any sense of pride or entitlement at the door.
+    Everyone is working together and doing their best.
+    Be graceful, polite, and unassuming in all interaction with the host team.
+    - Feedback from the host team is not any kind of personal criticism to you.
+    Treat and value it as data to help your contribution become all the better.
+    - If the host team decides not to accept a contribution, don't take it personally.
```
[the remark this answers is not in the document]
Comment (rrrutledge) [reply contributor/outline.md]: Sure!  Added a line: `Work with them to understand their reasoning.`

File: contributor/outline.md [hunk 22]
```diff
@@ -1,64 +1,141 @@
...
+    Remember and respect that the host team has ultimate decision over the project.
+    They're the ones that are signing up to maintain it, after all!
+    Try to collaborate on an aspect of the contribution that you all agree on,
+    and look for another way to fill the rest of your needs.
+  
+## Mechanics of Contributing
```
[the remark this answers is not in the document]
Comment (rrrutledge) [reply contributor/outline.md]: We have some discussion of that in **The Contributor Ethos**.  Is that section a good spot for those ideas?

File: contributor/outline.md [hunk 23]
```diff
@@ -1,64 +1,141 @@
...
+
+  - This section contains practical advice and gotchas to remember when making an InnerSource contribution.
```
[the remark this answers is not in the document]
Comment (rrrutledge) [reply contributor/outline.md]: Sounds fun.

File: contributor/outline.md [hunk 24]
```diff
@@ -1,64 +1,141 @@
...
+  - Preparing to work.
+    - Give yourself enough lead time.
+  The development, review, and deployment process for your contribution will likely take longer than similar processes on your own team.
+  Keep that in mind and plan to start early enough so that your work is available for you to leverage at the time you need it.
+    - If the host team is in a different time zone then keep that in mind and plan for it when interacting with them.
+    - For larger contributions, get agreement from the host team on the user/technical design of the contribution _before_ submitting a pull request.
+  This type of high-level, up-front agreeemment early will save time in rework or rejection of your pull request down the road.
+    - Understand how the host team would like you to reach out for help and questions while you're working.
+  - Development
+    - If you get stuck, try to unblock yourself by reading the project documentation or looking at other parts of the code in the project.
```
[the remark this answers is not in the document]
Comment (rrrutledge) [reply contributor/outline.md]: Added the word `first`.  The next sentence already talks about _after_ that then contacting the **Host** team for help.

File: contributor/outline.md [hunk 25]
```diff
@@ -1,64 +1,141 @@
...
+    - After trying the above if you're still stuck then reach out to the host team for help.
+    It doesn't do either side any good for you to spin forever on a problem that could be resolved in a few-minute conversation with someone more knowledgable about the project.
+    It's OK to ask for help.
+    You aren't expected to be as knowledgable and proficient as you might be on the full-time projects that your team owns.
+    - Match the existing coding style, indentation, etc, even if it's not what you would normally do.
```
[the remark this answers is not in the document]
Comment (rrrutledge) [reply contributor/outline.md]: Sounds good.

File: contributor/outline.md [hunk 26]
```diff
@@ -1,64 +1,141 @@
...
+    A guest contribution for a feature is not the time to introduce a new way of structuring or formatting project code.
+  - Submitting the pull request
+    - Run the project build and automated validation before opening a pull request.
```
[the remark this answers is not in the document]
Comment (rrrutledge) [reply contributor/outline.md]: I'm not sure - some projects may have ways to do test deployments/runs off of branches.  There could be some non-local steps to take before pull request.  Whatever is spelled out in the `README`/`CONTRIBUTING`.

File: contributor/outline.md [hunk 27]
```diff
@@ -1,64 +1,141 @@
...
+    Making pull requests too broad and large makes them more difficult to review and longer before they're accepted.
+  
+## Benefits of Contribution
+
+  - Personally
+    - Puts your development time into a place where others are also working.
+    It's not just up to you to maintain this area.
+    - Get help and support from the host team in implementing your changes.
+    - Your reputation and impact expands to outside your team.
+    - Learn from engineers on the host team and their coding and working styles.
```
[the remark this answers is not in the document]
Comment (rrrutledge) [reply contributor/outline.md]: Nice.

File: contributor/outline.md [hunk 28]
```diff
@@ -1,64 +1,141 @@
...
+    This extra learning comes as part of your day job - no stretch or overtime assignment needed.
+    - Varied, fun, and personally fulfilling.
+  - Team
+    - You can get improvements you need in other projects on a timeline that you control.
```
[the remark this answers is not in the document]
Comment (rrrutledge) [reply contributor/outline.md]: Changed to `you can influence`.  More control/influence than waiting for the **Host** team to get it done.

File: contributor/outline.md (hunk 6 above)
Comment (rrrutledge) [reply contributor/outline.md]: Good point.

File: contributor/outline.md (hunk 7 above)
Comment (rrrutledge) [reply contributor/outline.md]: _Reduced_

File: contributor/outline.md (hunk 8 above)
Comment (rrrutledge) [reply contributor/outline.md]: Is it good to have a low bus factor?  I think that [a high bus factor](https://en.wikipedia.org/wiki/Bus_factor#Increasing_the_bus_factor) is good.

I left it as `Improves` so that people can interpret as they like.

File: contributor/outline.md (hunk 9 above)
Comment (rrrutledge) [reply contributor/outline.md]: Well I think we could appeal the moral compass of teams by leaving it here.  The _Company_ section already calls out reduced duplication.

File: contributor/outline.md (hunk 10 above)
Comment (rrrutledge) [reply contributor/outline.md]: Done.

File: contributor/outline.md (hunk 11 above)
Comment (rrrutledge) [reply contributor/outline.md]: Yes.

File: contributor/outline.md (hunk 12 above)
Comment (rrrutledge) [reply contributor/outline.md]: 🤔.  Not sure.  Seems like more words to say the same thing?

File: contributor/outline.md (hunk 16 above)
[the remark this answers is not in the document]
Comment (guywmartin) [reply contributor/outline.md]: I agree with this, *however*, we need to be cognizant that getting inner source up and running in some orgs is more of a 'getting permissions' than 'being a renegade'.  I like @gruetter's proposed text for that reason.  While *I* like 'Just Do It(tm)', it doesn't mean it's universally applicable to all situations.

File: contributor/outline.md (hunk 8 above)
Comment (guywmartin) [reply contributor/outline.md]: Concur with 'Improves' in this context.

File: contributor/outline.md (hunk 16 above)
[the remark this answers is not in the document]
Comment (lenucksi) [reply contributor/outline.md]: I think this might depend upon the size of a contribution, i.e. as soon as efforts become larger and take up more time, convincing your management and possibly higher up levels of management becomes more important.
While your immediate line manager may be convinced, the entire thing may become infeasible if incentives are set in a way that would make line managers harm themselves on allowing contributions.

File: contributor/outline.md (hunk 12 above)
Comment (lenucksi) [reply contributor/outline.md]: The existing words have this as an implicit consequence, while @gruetter's words make this more explicit. 
Also everyone likes hearing about reduced waste, although I have heard various interpretations of what constitutes waste from various persons/roles... ;)

File: contributor/outline.md (hunk 16 above)
[the remark this answers is not in the document]
Comment (rrrutledge) [reply contributor/outline.md]: Recommend taking any suggestions here to the revised text (present in https://github.com/InnerSourceCommons/InnerSourceLearningPath/pull/64).
